packages/ckeditor5-widget/src/widget.ts

```diff
@@ -207,22 +207,19 @@ export default class Widget extends Plugin {
 			// But at least triple click inside nested editable causes broken selection in Safari.
 			// For such event, we select the entire nested editable element.
 			// See: https://github.com/ckeditor/ckeditor5/issues/1463.
-			if ( ( env.isSafari || env.isGecko ) && domEventData.domEvent.detail >= 3 ) {
-				const mapper = editor.editing.mapper;
-				const viewElement = element.is( 'attributeElement' ) ?
-					element.findAncestor( element => !element.is( 'attributeElement' ) )! : element;
-				const modelElement = mapper.toModelElement( viewElement )!;
-
-				domEventData.preventDefault();
 
-				this.editor.model.change( writer => {
-					writer.setSelection( modelElement, 'in' );
-				} );
+			if ( ( env.isSafari || env.isGecko ) && domEventData.domEvent.detail >= 3 ) {
+				setSelectionInAncestorElement();
 			}
 
 			return;
 		}
 
+		// If triple click should select entire paragraph.
+		if ( domEventData.domEvent.detail >= 3 ) {
+			setSelectionInAncestorElement();
+		}
```

Comment:
Maybe we should narrow it to just affected cases with widgets? This looks like first we check if in nested editable on FF or Safari and then we do it everywhere.